src/Illuminate/Foundation/Configuration/ApplicationBuilder.php

```diff
@@ -269,6 +269,18 @@ public function withMiddleware(?callable $callback = null)
             if ($priorities = $middleware->getMiddlewarePriority()) {
                 $kernel->setMiddlewarePriority($priorities);
             }
+
+            if ($priorityAppends = $middleware->getMiddlewarePriorityAppends()) {
+                foreach ($priorityAppends as $middleware => $after) {
+                    $kernel->addToMiddlewarePriorityAfter($after, $middleware);
+                }
+            }
+
+            if ($priorityPrepends = $middleware->getMiddlewarePriorityPrepends()) {
+                foreach ($priorityPrepends as $middleware => $before) {
+                    $kernel->addToMiddlewarePriorityBefore($before, $middleware);
+                }
+            }
```

Comment:
When using both `prependToPriorityList()` and `appendToPriorityList()` the `$middleware` variable is overridden on line 274. `getMiddlewarePriorityPrepends()` is then called on a string.

Reply:
Ah this seems to be a bug either prepending!